tests/handlers/auth_test.py

```diff
@@ -0,0 +1,358 @@
+        "/auth",
+        params=[("capability", "exec:test"), ("capability", "exec:admin")],
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 403
+    body = await r.text()
+    assert "No Capability group found in user's `isMemberOf`" in body
+    assert r.headers["X-Auth-Request-Token-Capabilities"] == "exec:test"
+    assert (
+        r.headers["X-Auth-Request-Capabilities-Accepted"]
+        == "exec:admin exec:test"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "all"
+
+
+async def test_authnz_token_success(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 200
+    assert (
+        r.headers["X-Auth-Request-Token-Capabilities"] == "exec:admin read:all"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Accepted"] == "exec:admin"
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "all"
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
+    assert r.headers["X-Auth-Request-User"] == "some-user"
+    assert r.headers["X-Auth-Request-Uid"] == "1000"
+    assert r.headers["X-Auth-Request-Groups"] == "admin"
+    assert r.headers["X-Auth-Request-Token"] == token
+    assert r.headers["X-Auth-Request-Token-Ticket"] == ""
+
+
+async def test_authnz_token_success_any(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["test"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params=[
+            ("capability", "exec:admin"),
+            ("capability", "exec:test"),
+            ("satisfy", "any"),
+        ],
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Token-Capabilities"] == "exec:test"
+    assert (
+        r.headers["X-Auth-Request-Capabilities-Accepted"]
+        == "exec:admin exec:test"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "any"
+    assert r.headers["X-Auth-Request-Groups"] == "test"
+
+
+async def test_authnz_token_forwarded(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={
+            "Authorization": "Basic blah",
+            "X-Forwarded-Access-Token": token,
+        },
+    )
+    assert r.status == 200
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
```

Comment:
So this test proves that the auth header isn't used by the forwarded token is used?

Reply:
Yes, indeed. Added a comment.